### ISSUE TITLE
more concrete error message for …

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#3511]
I wonder if this error message should be shown like that, it contains no information about the code, except that the vale it tried to concatenate was nil, e.g. it doesn't mention the line.
![nil](https://user-images.githubusercontent.com/21158813/44827672-6cd4aa00-ac14-11e8-8c38-7bdc1406a094.png)
How did that happen? Using [set_variable][join], while the last array element was incomplete:
[array][0]type,name=foo,bar
[array][1]type,name=foo,bar
[array][2]type=foo

**Comment by Pentarctagon** [comment]
Lua error messages are generally less helpful than WML messages, really.  There's also almost always some part of the stack trace that gets into things like `lua/wml-utils.lua` above, which is almost always just noise.

**Comment by sevu** [comment]
@CelticMinstrel do you have a comment on this? If it's not a trivial thing to change something here, please close this issue. It's probably fine as it is, and situational that it was not concrete in the above case.

**Comment by gfgtdf** [comment]
this is a bug in the lua implementaion and a regression form 1.12 (where this was implemented in c++), in 1.12 missing keys that liek were just treated the same as empty strings. It also trivial to fix in particular since the errromessage above does contain the line numbers.

**Comment by CelticMinstrel** [comment]
Yeah, I think fixing this in the tag implementation would be best.
